### ISSUE TITLE
[SLP] Assertion `!Mask.empty() && Reuses.size() == Mask.size() && "Expected non-empty mask."' failed.

### ISSUE THREAD

**Issue by d0k** [llvm/llvm-project#59230]
```ll
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

define void @main(ptr %0) {
  %.unpack = load double, ptr %0, align 8
  %.elt1 = getelementptr { double, double }, ptr %0, i64 0, i32 1
  %.unpack2 = load double, ptr %.elt1, align 8
  %2 = fadd double %.unpack, 0.000000e+00
  %3 = fsub double 0.000000e+00, %.unpack2
  %4 = fmul double %2, 5.000000e-01
  %5 = call double @llvm.fabs.f64(double %4)
  %6 = fmul double %3, 0.000000e+00
  %7 = call double @llvm.fabs.f64(double %6)
  %8 = fmul double %3, 5.000000e-01
  %9 = call double @llvm.fabs.f64(double %8)
  %10 = fmul double %2, 0.000000e+00
  %11 = call double @llvm.fabs.f64(double %10)
  %12 = fcmp oeq double %5, 1.000000e+00
  %13 = fcmp oeq double %7, 1.000000e+00
  %14 = or i1 %12, %13
  %15 = fcmp oeq double %11, 1.000000e+00
  %16 = or i1 %14, %15
  %17 = fcmp oeq double %9, 1.000000e+00
  %18 = or i1 %16, %17
  %19 = select i1 %18, double 0.000000e+00, double 1.000000e+00
  store double %19, ptr %0, align 8
  ret void
}

; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none)
declare double @llvm.fabs.f64(double) #0

; uselistorder directives
uselistorder ptr @llvm.fabs.f64, { 3, 2, 1, 0 }

attributes #0 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) }
```

```console
$ opt -passes=slp-vectorizer reduced.ll -mcpu=skx

opt: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:3530: void reorderReuses(SmallVectorImpl<int> &, ArrayRef<int>): Assertion `!Mask.empty() && Reuses.size() == Mask.size() && "Expected non-empty mask."' failed.
```

CC @alexey-bataev

**Comment by alexey-bataev** [comment]
Candidate patch 
https://reviews.llvm.org/D138819

**Comment by alexey-bataev** [comment]
Fixed in 0cc15050a423b1d1116677b3cd461a916ccf03fd
